log4j-core/src/main/java/org/apache/logging/log4j/core/appender/rolling/action/IfLastModified.java

```diff
@@ -79,35 +79,62 @@ public boolean accept(final Path basePath, final Path relativePath, final BasicF
         return result;
     }
 
-    /*
-     * (non-Javadoc)
-     *
-     * @see org.apache.logging.log4j.core.appender.rolling.action.PathCondition#beforeFileTreeWalk()
-     */
     @Override
     public void beforeFileTreeWalk() {
         IfAll.beforeFileTreeWalk(nestedConditions);
     }
 
     /**
-     * Create an IfLastModified condition.
-     *
-     * @param age The path age that is accepted by this condition. Must be a valid Duration.
-     * @param nestedConditions nested conditions to evaluate if this condition accepts a path
-     * @return An IfLastModified condition.
+     * @deprecated since 2.24.0 use {@link #newBuilder()}  instead.
      */
-    @PluginFactory
+    @Deprecated
     public static IfLastModified createAgeCondition(
-            // @formatter:off
-            @PluginAttribute("age") @Required(message = "No age provided for IfLastModified") final Duration age,
-            @PluginElement("PathConditions") final PathCondition... nestedConditions) {
-        // @formatter:on
-        return new IfLastModified(age, nestedConditions);
+            final org.apache.logging.log4j.core.appender.rolling.action.Duration age,
+            final PathCondition... pathConditions) {
+        return newBuilder()
+                .setAge(Duration.ofMillis(age.toMillis()))
+                .setNestedConditions(pathConditions)
+                .build();
     }
 
     @Override
     public String toString() {
         final String nested = nestedConditions.length == 0 ? "" : " AND " + Arrays.toString(nestedConditions);
         return "IfLastModified(age=" + age + nested + ")";
     }
+
+    /**
+     * @since 2.24.0
+     */
+    @PluginBuilderFactory
+    public static Builder newBuilder() {
+        return new Builder();
+    }
+
+    /**
+     * @since 2.24.0
+     */
+    public static final class Builder implements org.apache.logging.log4j.core.util.Builder<IfLastModified> {
+        @PluginBuilderAttribute
+        @Required(message = "No age provided for IfLastModified")
+        private org.apache.logging.log4j.core.appender.rolling.action.Duration age;
+
+        @PluginElement("nestedConditions")
+        private PathCondition[] nestedConditions;
+
+        public Builder setAge(final Duration age) {
```

Comment:
Shouldn't this be spelled as `withAge()` to be more consistent with other Builders syntax?

Reply:
Usually a `with*` method is provided by _immutable_ objects and creates a new instance of the object, so `set*` is more appropriate here.
In `2.x` you have a mixture of "setters" and "withers" and if both are present, the "wither" is **deprecated**.

The consensus is to replace all "withers" with "setters" in `3.x` (see #1206).

Reply:
Ok, understood.
Was just confused because other nearby Builder's already used the `withXyz()` syntax, so it seemed inconsistent when I encountered it here.